src/components/v2/TextField/styles.ts

```diff
@@ -12,21 +11,38 @@ export const useStyles = () => {
 
       ${hasError && `color: ${theme.palette.error.main};`};
     `,
-    getInputContainer: ({ hasError }: { hasError: boolean }) => css`
-      display: flex;
-      align-items: center;
-      padding: ${theme.spacing(2, 2, 2, 4)};
-      border-radius: ${theme.spacing(3)};
-      border: ${theme.spacing(0.5)} solid
-        ${hasError ? theme.palette.interactive.error : 'transparent'};
-      background-color: ${theme.palette.background.default};
+    getInputContainer: ({
+      hasError,
+      disabled,
+    }: {
+      hasError: boolean;
+      disabled: boolean | undefined;
+    }) => {
+      let borderColor: undefined | string = 'transparent';
```

Comment:
Nit: `borderColor` is never undefined

Reply:
According to our theme typing `theme.palette.interactive.error` could be undefined